### ISSUE TITLE
option to use mkvmerge instead of ffmpeg for merging

### ISSUE THREAD

**Issue by bacon-cheeseburger** [yt-dlp/yt-dlp#1879]
### Checklist

- [X] I'm reporting a feature request
- [X] I've verified that I'm running yt-dlp version **2021.12.01**. ([update instructions](https://github.com/yt-dlp/yt-dlp#update))
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

Hi. I just started using yt-dlp on a headless Linux system and the ffmpeg requirement for merging pulls in a ton of useless dependencies/bloat for headless setups. Is there any chance having the option to merge with mkvmerge (part of the mkvtoolnix package) instead would be considered?

Thanks

**Comment by nyuszika7h** [comment]
I've used ffmpeg on headless servers fine for ages. If you don't want to install a bunch of dependencies globally you can use johnvansickle's static builds.

But in general I do agree, support for mkvmerge would be nice, as generally it handles MKV muxing better.

**Comment by bacon-cheeseburger** [comment]
It's not that ffmpeg wouldn't work headless, it's the over 100+ useless dependencies for a headless system. That's a lot of bloat and potential breakage just to mux. Using a static build like you mentioned is a valid alternative but still not a great one given that mkvmerge (which I already have installed) does do mkv better than ffmpeg. I didn't mention that before so as to avoid any raging in case others here are very stubbornly pro-ffmpeg, which I've seen happen elsewhere.

I'm not a python guy but could probably copy postprocessor/ffmpeg.py to mkvmerge.py and mutilate it enough for hacky mkvmerge muxing but I know that's neither a great approach nor a PR I'd imagine would ever be accepted, not that I'd have the nerve to submit that anyway. I was hoping for something better, preferably done correctly by someone who actually knows what they're doing.

**Comment by Ashish0804** [comment]
No harm in giving a try and making a PR.
If it's lacking anything then it can be covered during the code review process 

See https://github.com/yt-dlp/yt-dlp/pull/757

**Comment by pukkandan** [comment]
> could probably copy postprocessor/ffmpeg.py to mkvmerge.py and mutilate it enough for hacky mkvmerge muxing but I know that's neither a great approach

Yeah, definitely not a great approach. The ffmpeg.py has a lot of stuff needed for the multitude of postprocessing tasks. Since the request is specifically about merging (and I assume you dont want to implement the other PPs with ffmpeg - if it is even possible) better to have FFmpegMergerPP to look for mkvmerge.exe (if the merging to mkv) and use it if available (See AtomicParsley implementation in EmbedThumbnailPP for a similar case).

This feature has been (unofficially) requested before. I personally think this is a redundant feature because (1) it only helps for mkv (2) only for merger and not other PPs and (3) ffmpeg does the job well enough already. So I am unwilling to work on this myself. But PRs on the other hand are welcome

**Comment by jayden000** [comment]
+1
Being able to use MKVMerge rather than ffmpeg would be very useful as ffmpeg doesn't passover some metadata from the original files such as, bitrate info for each thing in the container (video and audio). Hopefully someone could make a PR to implement.

**Comment by coreyoilar** [comment]
+1

**Comment by Kicer86** [comment]
Also, according to my observations, ffmpeg improperly embeds thumbnail image (for example see [here](https://superuser.com/questions/1430523/ffmpeg-thumbnail-not-added-and-filesize-increased-too-much-after-using-disposit))
